compiler/pipes/preprocess-function.cpp

```diff
@@ -16,6 +16,21 @@
 #include "compiler/name-gen.h"
 #include "compiler/type-hint.h"
 
+namespace {
+
+bool implicit_cast_allowed(bool strict_types, const TypeHint *type) {
```

Comment:
please, consider the following example

```php
<?php

/** @kphp-strict-types-enable */
declare(strict_types=1);


class A {
  public $x = 1;
}

function test_regexp_crap($re) {
  var_dump(preg_match($re, '123') > 0);
  var_dump(preg_match_all($re, '123') > 0);
}

test_regexp_crap(tuple(1));
```

before this patch kphp2cpp generated a compilation error
```
Starting php to cpp transpiling...

~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Compilation error at stage: Calc actual calls, gen by type-inferer.cpp:53
  dev.php:12  in test_regexp_crap
    var_dump(preg_match($re, '123') > 0);

pass tuple(int) to argument $regex of preg_match
but it's declared as @param regexp

  dev.php:12  in test_regexp_crap
    var_dump(preg_match($re, '123') > 0);
    $re is tuple(int)

  dev.php:11  in test_regexp_crap
    function test_regexp_crap($re) {
    argument $re inferred as tuple(int)

  dev.php:16
    
    tuple<> is tuple(int)
    1 is int


Compilation terminated due to errors
```

but with this patch kphp2cpp skips this error and generates C++ code which leads a gcc compilation error
```
test_regexp_crap.cpp:11:74: error: no matching function for call to ‘f$preg_match(const std::tuple<long int>&, string&)’
   11 |   f$var_dump(0_i64 < f$preg_match(v$re, v$const_string$us8c039e9e9dfbadf3));
```

Reply:
Fixed the tuple->regexp conversion.